metrics/ci_integrations/test/destination/firestore/adapter/firestore_destination_client_adapter_test.dart

```diff
@@ -79,20 +79,21 @@ void main() {
     );
 
     test(
-      ".addBuilds() returns normally if firestore throws GrpcError.notFound",
+      ".addBuilds() throws an exception if the firestore throws exception different from GrpcError.notFound",
       () {
-        whenFetchProject().thenThrow(GrpcError.notFound());
+        whenFetchProject().thenThrow(Exception());
 
         final result = adapter.addBuilds(testProjectId, []);
 
-        expect(result, completes);
+        expect(result, throwsException);
       },
     );
 
     test(
-      ".addBuilds() throws an exception if the firestore throws exception different from GrpcError.notFound",
+      ".addBuilds() throws an exception if a project with the given id does not exist",
       () {
-        whenFetchProject().thenThrow(Exception());
+        whenFetchProject().thenAnswer((_) => Future.value(_documentMock));
+        when(_documentMock.exists).thenReturn(false);
 
         final result = adapter.addBuilds(testProjectId, []);
 
```

Comment:
There is a test case below that isn't actual anymore (`.addBuilds() rethrows GrpcError with status different from 'notFound'`). Please rewrite the description or consider changing a test case.